### DISCUSSION TITLE
Preferred/sticky port numbers possible?

### DISCUSSION THREAD

**Discussion by vyruz1986** [testcontainers/testcontainers-dotnet#1229] [Q&A]
I really like using Testcontainers for integration tests, thanks for all the effort!

One thing I'm often struggling with, when quickly iterating over a test multiple times in order to debug and make the test succeed, sometimes I want to look into the 3rd party system I'm running with Testcontainers (e.g. SQL database, checking which records are created). Since by default Testcontainers uses a random port to expose the given service, I'll be continuously modifying my SQL client configuration to connect to the correct database port.

I'm aware we can start containers on a specific port, but that would cause problems in normal pipeline test runs where multiple tests might be running in parallel.

Is there a way to define a preferred or 'sticky' port number, that will be used when available, but have Testcontainers just fall back to its normal operation mode of choosing a random port whenever this preferred port is not available?

**Comment by 4dri4ng** [comment]
Hi,
try to use WithPortBinding:
`    private readonly MsSqlContainer _dbContainer = new MsSqlBuilder().WithPortBinding("63333","1433").Build();`

**Reply by vyruz1986** [reply]
Thanks for responding, but as I feared, this will throw an error if the given port is still in use. When running tests in parallel, this can cause problems, causing tests to fail
